src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/integrationtest/IntegrationTestUtils.java

```diff
@@ -0,0 +1,187 @@
+    }
+
+    /**
+     * Helper method to perform a submitted callback request.
+     *
+     * @param mockMvc      the MockMvc instance to use for the request
+     * @param jsonResource the JSON resource file to use as the request body
+     * @return the ResultActions of the performed request
+     * @throws Exception if an error occurs during the request
+     */
+    public static ResultActions performSubmittedCallback(MockMvc mockMvc, String jsonResource) throws Exception {
+        return performCallback(mockMvc, SUBMITTED_URL, jsonResource);
+    }
+
+    private static ResultActions performCallback(MockMvc mockMvc, String url, String jsonResource) throws Exception {
+        String request = getJsonFromFile(jsonResource);
+
+        return mockMvc.perform(post(url)
+            .content(request)
+            .header(AUTHORIZATION_HEADER, AUTH_TOKEN)
+            .contentType(MediaType.APPLICATION_JSON));
+    }
+
+    /**
+     * Helper method to mock the AssignCaseAccessService to return specified case roles for solicitors.
+     *
+     * @param assignCaseAccessService the AssignCaseAccessService to mock
+     * @param caseRoles               the case roles to return for the solicitors
+     */
+    public static void givenSolicitorsHaveCaseAccess(AssignCaseAccessService assignCaseAccessService, CaseRole... caseRoles) {
+        List<CaseAssignmentUserRole> caseAssignmentUserRoles = Arrays.stream(caseRoles)
+            .map(caseRole -> CaseAssignmentUserRole.builder()
+                .userId(UUID.randomUUID().toString())
+                .caseRole(caseRole.getCcdCode())
+                .build())
+            .toList();
+
+        CaseAssignmentUserRolesResource caseAssignmentUserRolesResource = CaseAssignmentUserRolesResource.builder()
+            .caseAssignmentUserRoles(caseAssignmentUserRoles)
+            .build();
+
+        when(assignCaseAccessService.getUserRoles(INTEGRATION_TEST_CASE_ID))
+            .thenReturn(caseAssignmentUserRolesResource);
+    }
+
+    /**
+     * Helper method to mock the CaseAssignedRoleService to return a specified role for the current user.
+     *
+     * @param caseAssignedRoleService the CaseAssignedRoleService to mock
+     * @param currentUserCaseRole     the case role to return for the current user
+     */
+    public static void givenCurrentUserHasRole(CaseAssignedRoleService caseAssignedRoleService, CaseRole currentUserCaseRole) {
+        if (currentUserCaseRole == CaseRole.CASEWORKER) {
+            mockCurrentUserCaseAccess(caseAssignedRoleService, Collections.emptyList());
+        } else {
+            List<CaseAssignedUserRole> caseAssignedUserRoles = List.of(
+                CaseAssignedUserRole.builder()
+                    .userId(UUID.randomUUID().toString())
+                    .caseRole(currentUserCaseRole.getCcdCode())
+                    .build());
+            mockCurrentUserCaseAccess(caseAssignedRoleService, caseAssignedUserRoles);
+        }
+    }
```

Comment:
Haven't quite sussed why Caseworker is an exception.  I could be missing something obvious.  If not, maybe a comment?

Reply:
A caseworker is never assigned an explicit case role for a case whereas all professional users are. So the applicant solicitor has [APPSOLICITOR] and applicant barrister has [APPBARRISTER] and so on. So the API returns an empty collection for caseworker and a collection of 1 for other users. See the case_users table in your local db.